components/concept/engine/src/main/java/mozilla/components/concept/engine/prompt/PromptRequest.kt

```diff
@@ -29,37 +29,52 @@ sealed class PromptRequest(
      * Value type that represents a request for a single choice prompt.
      * @property choices All the possible options.
      * @property onConfirm A callback indicating which option was selected.
+     * @property onDismiss A callback executed when dismissed.
      */
-    data class SingleChoice(val choices: Array<Choice>, val onConfirm: (Choice) -> Unit) : PromptRequest()
+    data class SingleChoice(
+        val choices: Array<Choice>,
+        val onConfirm: (Choice) -> Unit,
+        override val onDismiss: () -> Unit
+    ) : PromptRequest(), Dismissible
 
     /**
      * Value type that represents a request for a multiple choice prompt.
      * @property choices All the possible options.
      * @property onConfirm A callback indicating witch options has been selected.
+     * @property onDismiss A callback executed when dismissed.
      */
-    data class MultipleChoice(val choices: Array<Choice>, val onConfirm: (Array<Choice>) -> Unit) : PromptRequest()
+    data class MultipleChoice(
+        val choices: Array<Choice>,
+        val onConfirm: (Array<Choice>) -> Unit,
+        override val onDismiss: () -> Unit
+    ) : PromptRequest(), Dismissible
 
     /**
      * Value type that represents a request for a menu choice prompt.
      * @property choices All the possible options.
      * @property onConfirm A callback indicating which option was selected.
+     * @property onDismiss A callback executed when dismissed.
      */
-    data class MenuChoice(val choices: Array<Choice>, val onConfirm: (Choice) -> Unit) : PromptRequest()
+    data class MenuChoice(
+        val choices: Array<Choice>,
+        val onConfirm: (Choice) -> Unit,
+        override val onDismiss: () -> Unit
+    ) : PromptRequest(), Dismissible
 
     /**
      * Value type that represents a request for an alert prompt.
      * @property title of the dialog.
      * @property message the body of the dialog.
      * @property hasShownManyDialogs tells if this page has shown multiple prompts within a short period of time.
-     * @property onDismiss callback to let the page know the user dismissed the dialog.
      * @property onConfirm tells the web page if it should continue showing alerts or not.
+     * @property onDismiss callback to let the page know the user dismissed the dialog.
      */
     data class Alert(
         val title: String,
         val message: String,
         val hasShownManyDialogs: Boolean = false,
-        override val onDismiss: () -> Unit,
-        val onConfirm: (Boolean) -> Unit
+        val onConfirm: (Boolean) -> Unit,
+        override val onDismiss: () -> Unit
```

Comment:
@emilio is there any specific reason why we are changing the order of the parameters? This will cause a breaking change for consumer and it could cause that consumers that don't use named parameter to invert the functionality of `onConfirm` and onDismiss.

Reply:
If we don't have an strong reason, I recommend avoid changing the ordering of the parameters

Reply:
I talked with @Amejia481 about this. I think it's fine, but I would update the changelog to warn of breaking changes.

Reply:
@Amejia481 well, there are some parameters that are `onConfirm`, then `onDismiss`, and others which are the other way around. I didn't know which way to go when adding these so added them at the end since it felt a bit more natural to me, and then added a second commit updating the order to be consistent.

I don't strongly care about this, so for now per the above I'll update the changelog, but let me know if you want me to drop the second commit entirely or what not.

Reply:
As @gabrielluong mentioned I think with the the changelog entry should be sufficient. Thanks for making the update.